core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -988,12 +997,16 @@ private[spark] object JsonProtocol {
     rddInfo
   }
 
-  def storageLevelFromJson(json: JValue): StorageLevel = {
-    val useDisk = (json \ "Use Disk").extract[Boolean]
-    val useMemory = (json \ "Use Memory").extract[Boolean]
-    val deserialized = (json \ "Deserialized").extract[Boolean]
-    val replication = (json \ "Replication").extract[Int]
-    StorageLevel(useDisk, useMemory, deserialized, replication)
+  def storageLevelFromJson(json: JValue): StorageLevel = json match {
+    case _: JString => StorageLevel.fromString(json.extract[String])
```

Comment:
It's probably worth some comments about why there are two read paths

Reply:
Done.